### PR TITLE
Handle when namespace not updated

### PR DESCRIPTION
**Pull request by sedroche**
## Additional Information
QE build is failing => https://integreatly-qe-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/view/all/job/installation-pipeline-qe-pony/10/console
- Adding more refined error handling
- Adding label for uninstall as per https://github.com/integr8ly/installation/pull/322

### CODE REVIEW

File: evals/roles/kube_state_metrics/tasks/main.yml [hunk 1]
```diff
@@ -4,8 +4,11 @@
   register: create_namespace
   failed_when: create_namespace.stderr != '' and 'AlreadyExists' not in create_namespace.stderr
 
-- name: Add monitoring label to namespace
-  shell: oc patch ns {{ kube_state_metrics_namespace }} --patch '{"metadata":{"labels":{"{{ monitoring_label_name }}":"{{ monitoring_label_value }}"}}}'
+- name: Add labels to namespace
+  shell: oc patch ns {{ kube_state_metrics_namespace }} --patch '{"metadata":{"labels":{"{{ monitoring_label_name }}":"{{ monitoring_label_value }}", "integreatly-middleware-service":"true"}}}'
```
Comment (pb82) [inline evals/roles/kube_state_metrics/tasks/main.yml]: Any reason why not just use `oc label namespace` here?

Comment (sedroche) [reply evals/roles/kube_state_metrics/tasks/main.yml]: It does seem a safer option. @philbrookes is there a reason?

Comment (philbrookes) [reply evals/roles/kube_state_metrics/tasks/main.yml]: no, I was just reusing a command I found elsewhere, happy to switch to `oc label namespace`!

Comment (sedroche) [reply evals/roles/kube_state_metrics/tasks/main.yml]: I can update then

Comment (sedroche) [reply evals/roles/kube_state_metrics/tasks/main.yml]: have updated this but need to verify @pb82

Comment (sedroche) [reply evals/roles/kube_state_metrics/tasks/main.yml]: @pb82 install was a success => @pb82 https://integreatly-qe-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/view/all/job/installation-pipeline-qe-pony/11/

```bash
oc describe namespace middleware-monitoring
Name:         middleware-monitoring
Labels:       integreatly-middleware-service=true
              monitoring-key=middleware
```

**Pull request merged**